app/models/v2/report.rb

```diff
@@ -7,9 +7,45 @@ class Report < ApplicationRecord
     self.table_name = :reports
     self.primary_key = :id
 
+    SYSTEM_COUNT = lambda do
+      AN::NamedFunction.new(
+        'COUNT', [V2::System.arel_table[:id]]
+      ).filter(Pundit.policy_scope(User.current, V2::System).arel.ast.cores.first.wheres.first)
```

Comment:
I think here you can experiment with the `where_clause.ast` and the `.first` at the end is definitely unsafe.